### PR TITLE
feat: cohorts, data sources and variables from nested networks are also included in current network views

### PR DESCRIPTION
**Pull request by mswertz**
closes https://github.com/molgenis/GCC/issues/521

What are the main changes you did:
- added partOfNetworks refback to get what networks a network is part of
- used that to in graphql queries to include cohorts, variables, data sources in counts and lists

how to test:
- use the networkOfnetworks and compare results before and after; you should see +1 cohorts, variables to be included.

todo:
- [x] add e2e test for these specific changes
- [x] variable counts are correct but link to variable harmonisation details is broken in variables in nested network

### CODE REVIEW

File: apps/nuxt3-ssr/pages/[schema]/ssr-catalogue/[catalogue]/cohorts/index.vue [hunk 1]
```diff
@@ -150,7 +150,21 @@ const gqlFilter = computed(() => {
 
   // add hard coded page sepsific filters
   if ("all" !== route.params.catalogue) {
-    result["networks"] = { id: { equals: route.params.catalogue } };
+    result = {
```
Comment (harmbrugge) [inline apps/nuxt3-ssr/pages/[schema]/ssr-catalogue/[catalogue]/cohorts/index.vue:153]: This filter (the or part at least) is repeated 4 times. Maybe make this a formatted string, if that's a thing in java script.

**Review by harmbrugge — CHANGES_REQUESTED** [review]

File: apps/nuxt3-ssr/pages/[schema]/ssr-catalogue/[catalogue]/variables/index.vue [hunk 2]
```diff
@@ -236,7 +243,10 @@ async function buildScopedModelFilter() {
     return { error };
   }
 
-  const modelIds = data.Networks[0].models.map((m: { id: string }) => m.id);
+  console.log(JSON.stringify(data.Networks));
```
[the remark this answers is not in the document]
Comment (mswertz) [reply apps/nuxt3-ssr/pages/[schema]/ssr-catalogue/[catalogue]/variables/index.vue:246]: indeed

File: apps/nuxt3-ssr/pages/[schema]/ssr-catalogue/[catalogue]/variables/index.vue [hunk 3]
```diff
@@ -281,7 +291,14 @@ const fetchData = async () => {
               ...filter.value,
               ...{
                 networkVariables: {
-                  network: { id: { equals: catalogueRouteParam } },
+                  _or: [
+                    { network: { id: { equals: catalogueRouteParam } } },
```
[the remark this answers is not in the document]
Comment (mswertz) [reply apps/nuxt3-ssr/pages/[schema]/ssr-catalogue/[catalogue]/variables/index.vue:295]: indeed, this might be the bug. Should be conditional on !='all'

**Comment by mswertz** [comment]
> please update tests as needed

@EleanorHyde-UMCG can you assign somebody for the test? I will move the PR to 'catalogue' team

**Review by harmbrugge — APPROVED** [review]

**Comment by mswertz** [comment]
> Code looks good , 2 small changes / checks requested, and a test is needed for regression. On the model side i'm a bit worried about the number of checks and joins needed for basis functions might be a sign of issue in the way the domain is modelled.
> 
> * see comments
> * add test

Agree on model but I think it is as simple as we can.
Test is an interesting one. I would like @EleanorHyde-UMCG to add an acceptance test and we can add e2e tests. How can I do that most easily?

**Comment by EleanorHyde-UMCG** [comment]
First of all could someone (@mswertz?) please confirm that the following is a correct description of the query?

_All variables in a network-of-networks are: all the variables in all the CDMs which have been designated as belonging to the networks hanging under the network-of-networks in question PLUS any variables which have been designated as specifically belonging to the network-of-networks (for whatever reason).  With duplicates removed._

**Review by BrendaHijmans — CHANGES_REQUESTED** [review]
Functional review: variable counts are correct but link to variable harmonisation details is broken in variables in nested network. To test, upload this zip in catalogue-demo schema on preview: 
[catalogue-demo1721292572235.zip](https://github.com/user-attachments/files/16280345/catalogue-demo1721292572235.zip)
Go to EHEN network, click on vaiables, click on a variable, get an empty blue screen.

File: data/_models/shared/DataCatalogue-TODO.csv
[the remark this answers is not in the document]
Comment (mswertz) [reply data/_models/shared/DataCatalogue-TODO.csv:1]: done

File: apps/nuxt3-ssr/pages/[schema]/ssr-catalogue/[catalogue]/variables/index.vue (hunk 3 above)
[the remark this answers is not in the document]
Comment (mswertz) [reply apps/nuxt3-ssr/pages/[schema]/ssr-catalogue/[catalogue]/variables/index.vue:297]: this is a 'scoped' block

**Review by BrendaHijmans — CHANGES_REQUESTED** [review]
The variable count on the network homepage is not correct. Upload this excel in catalogue_demo to test: 
[var_counts_test.xlsx](https://github.com/user-attachments/files/16310762/var_counts_test.xlsx). Go to TESTNETWORK in ssr to test. Variable count should be 25 (10 from test1, 10 from test3 and 5 from test2 via Network variables). Also, the link to variable details is broken. See pictures for discrepancy in counts, counts on hompepage are not correct, but list count is.
image
image

**Comment by mswertz** [comment]
> code looks good , please remove console.log statements

done

**Comment by mswertz** [comment]
> The variable count on the network homepage is not correct. Upload this excel in catalogue_demo to test:
> [var_counts_test.xlsx](https://github.com/user-attachments/files/16310762/var_counts_test.xlsx). Go to TESTNETWORK in ssr to test.

fixed. Turned out that this network has no cohorts and thus no mappings. Added check for empty cohorts list.

**Pull request merged**